tempesta_fw/ss_skb.c

```diff
@@ -697,12 +715,13 @@ __skb_fragment(struct sk_buff *skb, char *pspt, int len, TfwStr *it)
 	 * advance the skb tail pointer.
 	 */
 	if (len > 0) {
+		if (unlikely((offset == d_size)))
+			pspt = (char *)skb_frag_address(&si->frags[0]);
 		offset = pspt - (char *)skb_frag_address(&si->frags[0]);
```

Comment:
Basically it's just equal to

         offset = (unlikely((offset == d_size))) ? 0 : pspt - (char *)skb_frag_address(&si->frags[0]);

So there is no need to calculate `pspt` just to get zero at next line.